x/emissions/keeper/keeper.go

```diff
@@ -1072,17 +1137,102 @@ func (k *Keeper) GetLatestTopicInferences(ctx context.Context, topicId TopicId)
 	}
 	defer iter.Close()
 
+	inferences := &types.Inferences{
+		Inferences: make([]*types.Inference, 0),
+	}
+	var blockHeight int64 = 0
+
 	if iter.Valid() {
 		keyValue, err := iter.KeyValue()
 		if err != nil {
 			return nil, 0, errorsmod.Wrap(err, "error getting key value")
 		}
-		return &keyValue.Value, keyValue.Key.K2(), nil
+		inferences = &keyValue.Value
+		blockHeight = keyValue.Key.K2()
+
+		if outlierResistant {
+			filteredInferences, err := k.FilterOutlierResistantInferences(ctx, topicId, *inferences)
+			if err != nil {
+				return nil, 0, errorsmod.Wrap(err, "error filtering outlier resistant inferences")
+			}
+			inferences = &filteredInferences
+		}
 	}
 
-	return &types.Inferences{
-		Inferences: make([]*types.Inference, 0),
-	}, 0, nil
+	return inferences, blockHeight, nil
+}
+
+// UpdateNetworkInferencesOutlierMetrics recalculates the network inferences outlier metrics
```

Comment:
I don't love the use of the word "metrics" here because at least to me, it implies some kind of offchain database that is tracking performance data or something.

I'm also just unclear what this function is about based on the function header description. It seems like it updates the "previous" MAD and median values

Reply:
> I don't love the use of the word "metrics" here

You got a point there on the metrics. Couldn't come up with a better name myself, so open to suggestions. "Data" also sound too vague and "stats" (maybe better?) also give the idea that they are not used in the logic. 

Natif Inglish speaka, halp!

> I'm also just unclear what this function is about based on the function header description. It seems like it updates the "previous" MAD and median values

Indeed. The _metrics_ used for outlier detection.